hypha/apply/users/models.py

```diff
@@ -153,14 +153,20 @@ def get_or_create_and_notify(self, defaults: dict | None =None, site=None, **kwa
             elif not user.is_active:
                 raise IntegrityError("Found an inactive account")
         else:
-            temp_pass = BaseUserManager().make_random_password(length=32)
-            temp_pass_hash = make_password(temp_pass)
+            # Coming from registration without application
+            if 'password' in kwargs:
+                temp_pass_hash = make_password(kwargs.pop('password'))
+            else:
+                temp_pass = BaseUserManager().make_random_password(length=32)
+                temp_pass_hash = make_password(temp_pass)
```

Comment:
```suggestion
            if 'password' in kwargs:
                # Coming from registration without application
                temp_pass = kwargs.pop('password')
            else:
                temp_pass = BaseUserManager().make_random_password(length=32)
                
            temp_pass_hash = make_password(temp_pass)
```

This can be simplified.